### DISCUSSION TITLE
RFE: Store auxiliary data in rpmdb

### DISCUSSION THREAD

**Discussion by j-mracek** [rpm-software-management/rpm#2869] [Ideas]
It is difficult to decide which package from a repository was installed, because repodata created by createrepo contains only checksum of the whole package and rpmdb contains only checksum of a header. What about to store such an information in RPMDB?

**Comment by dmach** [comment]
Multiple checksums associated with each package would be nice.
For example:
 * an old repo uses sha1 checksums
 * dnf uses sha256 checksum as an internal primary package identifier
 * dnf computes both checksums and stores them in rpmdb when a package gets installed/upgraded

**Comment by pmatilai** [comment]
We have way too many checksums in rpm as it is :sweat_smile: 

The header is the data by which a package should be identified, because the whole can differ due to uncompressed vs compressed payload, both of which are acceptable to rpm as long as the bits are identical. 

The generic case here would be permitting storage of auxiliary data in rpmdb, which has been requested for various purposes for as long as I remember. I think mostly everybody agrees that there are legitimate cases for it, but the devil is in the details.
